### PR TITLE
r/instance: prevent double encoding when updating `user_data` or `user_data_base64`

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.com/hashicorp/terraform-provider-aws/issues/23305
Relates https://github.com/hashicorp/terraform-provider-aws/issues/23315#issuecomment-1049832011

Output from acceptance testing:

```
--- PASS: TestAccEC2Instance_userDataBase64 (133.10s)
--- PASS: TestAccEC2Instance_userDataBase64_updateWithBashFile (170.21s)
--- PASS: TestAccEC2Instance_userDataBase64_update (174.85s)
--- PASS: TestAccEC2Instance_userDataBase64_updateWithZipFile (175.79s)

--- PASS: TestAccEC2Instance_UserData (100.12s)
--- PASS: TestAccEC2Instance_UserData_update (312.70s)
--- PASS: TestAccEC2Instance_UserData_emptyStringToUnspecified (128.04s)
--- PASS: TestAccEC2Instance_UserData_unspecifiedToEmptyString (129.84s)
--- PASS: TestAccEC2Instance_UserData_stringToEncodedString (196.40s)
```

### CODE REVIEW

File: internal/service/ec2/instance_test.go [hunk 1]
```diff
@@ -3583,6 +3651,39 @@ func TestAccEC2Instance_UserData_update(t *testing.T) {
 	})
 }
 
+func TestAccEC2Instance_UserData_stringToEncodedString(t *testing.T) {
```
Comment (anGie44) [inline internal/service/ec2/instance_test.go:3654]: this test might not be releastic?

**Comment by ewbankkit** [comment]
Relates https://github.com/hashicorp/terraform-provider-aws/pull/18043.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEC2Instance_userDataBase64\|TestAccEC2Instance_UserData\|TestAccEC2Instance_basic' PKG=ec2 ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 3  -run=TestAccEC2Instance_userDataBase64\|TestAccEC2Instance_UserData\|TestAccEC2Instance_basic -timeout 180m
=== RUN   TestAccEC2Instance_basic
=== PAUSE TestAccEC2Instance_basic
=== RUN   TestAccEC2Instance_userDataBase64
=== PAUSE TestAccEC2Instance_userDataBase64
=== RUN   TestAccEC2Instance_userDataBase64_updateWithBashFile
=== PAUSE TestAccEC2Instance_userDataBase64_updateWithBashFile
=== RUN   TestAccEC2Instance_userDataBase64_updateWithZipFile
=== PAUSE TestAccEC2Instance_userDataBase64_updateWithZipFile
=== RUN   TestAccEC2Instance_userDataBase64_update
=== PAUSE TestAccEC2Instance_userDataBase64_update
=== RUN   TestAccEC2Instance_UserData
=== PAUSE TestAccEC2Instance_UserData
=== RUN   TestAccEC2Instance_UserData_update
=== PAUSE TestAccEC2Instance_UserData_update
=== RUN   TestAccEC2Instance_UserData_stringToEncodedString
=== PAUSE TestAccEC2Instance_UserData_stringToEncodedString
=== RUN   TestAccEC2Instance_UserData_emptyStringToUnspecified
=== PAUSE TestAccEC2Instance_UserData_emptyStringToUnspecified
=== RUN   TestAccEC2Instance_UserData_unspecifiedToEmptyString
=== PAUSE TestAccEC2Instance_UserData_unspecifiedToEmptyString
=== CONT  TestAccEC2Instance_basic
=== CONT  TestAccEC2Instance_UserData
=== CONT  TestAccEC2Instance_UserData_emptyStringToUnspecified
--- PASS: TestAccEC2Instance_basic (96.48s)
=== CONT  TestAccEC2Instance_UserData_unspecifiedToEmptyString
--- PASS: TestAccEC2Instance_UserData (129.83s)
=== CONT  TestAccEC2Instance_userDataBase64_updateWithZipFile
--- PASS: TestAccEC2Instance_UserData_emptyStringToUnspecified (170.68s)
=== CONT  TestAccEC2Instance_userDataBase64_update
--- PASS: TestAccEC2Instance_UserData_unspecifiedToEmptyString (163.16s)
=== CONT  TestAccEC2Instance_UserData_stringToEncodedString
--- PASS: TestAccEC2Instance_userDataBase64_updateWithZipFile (215.12s)
=== CONT  TestAccEC2Instance_UserData_update
--- PASS: TestAccEC2Instance_userDataBase64_update (214.74s)
=== CONT  TestAccEC2Instance_userDataBase64_updateWithBashFile
--- PASS: TestAccEC2Instance_UserData_stringToEncodedString (235.36s)
=== CONT  TestAccEC2Instance_userDataBase64
--- PASS: TestAccEC2Instance_UserData_update (213.35s)
--- PASS: TestAccEC2Instance_userDataBase64 (116.52s)
--- PASS: TestAccEC2Instance_userDataBase64_updateWithBashFile (235.30s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	627.223s
```

**Comment by zrma** [comment]
I can't thank you enough for your help and contribution. 👍 🙇

**Pull request merged**